cleanlab/count.py

```diff
@@ -656,9 +667,24 @@ def estimate_confident_joint_and_cv_pred_proba(
         X_train_cv, X_holdout_cv = X[cv_train_idx], X[cv_holdout_idx]
         s_train_cv, s_holdout_cv = labels[cv_train_idx], labels[cv_holdout_idx]
 
-        # Fit the clf classifier to the training set and
-        # predict on the holdout set and update pred_probs.
-        clf_copy.fit(X_train_cv, s_train_cv)
+        # Ensure no missing classes in training set.
+        train_cv_classes = set(s_train_cv)
+        all_classes = set(range(K))  # TODO: more efficient if committed to labels 0,...,K, else use this: set(labels)
+        if len(train_cv_classes) != len(all_classes):
```

Comment:
When is this possible? Unless there is only a single data point of a particular class in the entire dataset, shouldn't every fold of the stratified k-fold cross val have at least one data point of each class in the training split? And if there's only a single data point of a particular class in the entire dataset, handling it in this way doesn't seem great; that single point will be in both the train and test set. Might cause downstream issues, too; the threshold for the class is going to be equal to the predicted probability for that data point, and then by this logic (https://github.com/cleanlab/cleanlab/blob/master/cleanlab/count.py#L388), there will be no data that ends up in that entry on the diagonal of the confident joint, and ...

(the consequences / fix might be easier to discuss in-person)

Reply:
Yes there is a duplicated copy of this one point in both train and test set (the test set copy is the only that matters though). The duplicate train set point is necessary to ensure model training does not fail (model's training set needs to have all classes for many model-types), and it should have minimal impact on the model.

I've updated the code to overwrite the model's prediction for the test set point. So now we use a hard-coded dummy-prediction for the duplicated test point in each crossval fold. For now, dummy prediction = one-hot vector with 1 in the entry for the given label. I've verified this at least allows LearningWithNoisyLabels to avoid crashing and does not introduce noticeable performance difference AFAICT.